### PR TITLE
Adds dynamic datastructure configuration code sample

### PR DESCRIPTION
**Pull request by vbekiaris**

### CODE REVIEW

**Comment by Donnerbart** [comment]
Did you check CeckStyle locally?

File: learning-basics/configure-datastructures-at-runtime/src/main/java/ClientDynamicConfig.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public static void main(String[] args) {
+        // Start a member with no explicit configuration
+        HazelcastInstance instance = Hazelcast.newHazelcastInstance();
+        ILogger logger = instance.getLoggingService().getLogger(ClientDynamicConfig.class);
+
+        // Start a client -- we will be using the map to access & configure data structures
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        // Obtain a map
+        IMap<String, String> defaultMap = client.getMap("defaultMap");
+        // no data have been added yet, get("1") will return null
```
Comment (Donnerbart) [inline learning-basics/configure-datastructures-at-runtime/src/main/java/ClientDynamicConfig.java]: `// no data has been added yet, so get("1") will return null`

**Comment by vbekiaris** [comment]
Surprisingly, checkstyle passes even without any license header on the new files. The problem with `checkstyle` profile was wrong definition of `main.basedir` property.

File: learning-basics/configure-datastructures-at-runtime/src/main/java/ClientDynamicConfig.java (hunk 1 above)
Comment (vbekiaris) [reply learning-basics/configure-datastructures-at-runtime/src/main/java/ClientDynamicConfig.java]: You may be interested in the "is data singular or plural" debate 😉  : https://www.theguardian.com/news/datablog/2010/jul/16/data-plural-singular

**Pull request merged**